python/tank/commands/push_pc.py

```diff
@@ -143,21 +238,33 @@ def run_interactive(self, log, args):
                         dev_desc = desc
                         break
         if dev_desc:
-            log.warning("Looks like you have one or more dev locations set up in your "
-                        "configuration! We strongly recommend that you do not use dev locations "
-                        "in any production based configs. Dev descriptors are for development "
-                        "purposes only. You can easily switch a dev location using the "
-                        "'tank switch_app' command.")
-            if not console_utils.ask_yn_question("Okay to proceed?"):
+            log.warning(
+                "Looks like you have one or more dev locations set up in your "
+                "configuration! We strongly recommend that you do not use dev locations "
+                "in any production based configs. Dev descriptors are for development "
+                "purposes only. You can easily switch a dev location using the "
+                "'tank switch_app' command."
+            )
+            # Assume "yes" in non interactive mode
+            if self._is_interactive and not console_utils.ask_yn_question("Okay to proceed?"):
                 raise TankError("Aborted.")
         
         date_suffix = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
         
         source_path = os.path.join(self.tk.pipeline_configuration.get_path(), "config")
-        target_tmp_path = os.path.join(target_pc_path, "config.tmp.%s" % date_suffix)
-        symlink_path = os.path.join(target_pc_path, "config.%s" % date_suffix)
+        # Protect ourself against an edge case which happens mostly in unit tests
+        # if multiple pushes are attempted within less than a second, which is the
+        # granularity of our date_suffix used for uniqueness.
+        target_tmp_path = filesystem.get_unused_path(
```

Comment:
I preserved the original code. I guess it's all about using user friendly temp file names, which can may be be achieved with `tempfile`, although I'm not sure it is possible to always avoid a cryptic bit in the resulting name...